test/end2end_test.go

```diff
@@ -419,13 +446,19 @@ func (e env) dialer(addr string, timeout time.Duration) (net.Conn, error) {
 }
 
 var (
-	tcpClearEnv   = env{name: "tcp-clear-v1-balancer", network: "tcp", balancer: "v1"}
-	tcpTLSEnv     = env{name: "tcp-tls-v1-balancer", network: "tcp", security: "tls", balancer: "v1"}
-	tcpClearRREnv = env{name: "tcp-clear", network: "tcp", balancer: "round_robin"}
-	tcpTLSRREnv   = env{name: "tcp-tls", network: "tcp", security: "tls", balancer: "round_robin"}
-	handlerEnv    = env{name: "handler-tls", network: "tcp", security: "tls", httpHandler: true, balancer: "round_robin"}
-	noBalancerEnv = env{name: "no-balancer", network: "tcp", security: "tls"}
-	allEnv        = []env{tcpClearEnv, tcpTLSEnv, tcpClearRREnv, tcpTLSRREnv, handlerEnv, noBalancerEnv}
+	tcpClearEnv       = env{name: "tcp-clear-v1-balancer", network: "tcp", balancer: "v1"}
+	tcpTLSEnv         = env{name: "tcp-tls-v1-balancer", network: "tcp", security: "tls", balancer: "v1"}
+	tcpLocalEnv       = env{name: "tcp-local-v1-balancer", network: "tcp", security: "local", balancer: "v1"}
+	tcpIPV6LocalEnv   = env{name: "tcp-ipv6-local-v1-balancer", network: "tcp", listenerAddr: "[::1]:0", security: "local", balancer: "v1"}
+	unixTLSEnv        = env{name: "unix-local-v1-balancer", network: "unix", security: "local", balancer: "v1"}
+	tcpClearRREnv     = env{name: "tcp-clear", network: "tcp", balancer: "round_robin"}
+	tcpTLSRREnv       = env{name: "tcp-tls", network: "tcp", security: "tls", balancer: "round_robin"}
+	tcpLocalRREnv     = env{name: "tcp-local", network: "tcp", security: "local", balancer: "round_robin"}
+	tcpIPV6LocalRREnv = env{name: "tcp-ipv6-local", network: "tcp", listenerAddr: "[::1]:0", security: "local", balancer: "round_robin"}
+	unixLocalRREnv    = env{name: "unix-local", network: "unix", security: "local", balancer: "round_robin"}
+	handlerEnv        = env{name: "handler-tls", network: "tcp", security: "tls", httpHandler: true, balancer: "round_robin"}
+	noBalancerEnv     = env{name: "no-balancer", network: "tcp", security: "tls"}
+	allEnv            = []env{tcpClearEnv, tcpTLSEnv, tcpLocalEnv, tcpIPV6LocalEnv, tcpClearRREnv, unixTLSEnv, tcpTLSRREnv, tcpLocalRREnv, tcpIPV6LocalRREnv, unixLocalRREnv, handlerEnv, noBalancerEnv}
```

Comment:
I believe extending this will significantly increase the runtime of the test.  I don't think we need every test case to run with local credentials - testing local credentials in one test should be sufficient.